lib/dpl/provider/pages.rb

```diff
@@ -109,25 +119,55 @@ def setup_auth
       end
 
       def check_auth
-        setup_auth
+        if @gh_token
+          begin
+            setup_auth
 
-        unless api.scopes.include? 'public_repo' or api.scopes.include? 'repo'
-          error "Dpl does not have permission to access #{@gh_url} using it. Make sure your token contains the repo or public_repo scope."
-        end
+            unless api.scopes.include? 'public_repo' or api.scopes.include? 'repo'
+              error "Dpl does not have permission to access #{@gh_url} using it. Make sure your token contains the repo or public_repo scope."
+            end
 
-        log "Logged in as @#{user.login} (#{user.name})"
-      rescue Octokit::Unauthorized => exc
-        error "gh-token is invalid. Details: #{exc}"
+            log "Logged in as @#{user.login} (#{user.name})"
+          rescue Octokit::Unauthorized => exc
+            error "gh-token is invalid. Details: #{exc}"
+          end
+        elsif  @gh_deploy_key
+          Dir.mktmpdir do |tmpdir|
+            github_install_key(tmpdir)
+            context.shell "ssh -T git@#{@gh_url}"
+            if $?.exitstatus == 255
+              error "deploy-key is invalid"
+            end
+          end
+        else
+          error "must specify github-token or deploy-key"
+        end
       end
 
       def needs_key?
+        # Provider's generic support for keys is about provisioning and
+        # deprovisioning them, while this provider needs to install a key
+        # provided by the caller.
         false
       end
 
       def print_step(msg)
         log msg if @verbose
       end
 
+      def github_install_key(tmpdir)
+        if @gh_deploy_key
```

Comment:
You already do this check when calling this method. I think it's not good to silence errors this way